geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreConfiguration.java

```diff
@@ -0,0 +1,82 @@
+     * @param name String representation of the id of the BlobStore to remove. This
+     * @throws NoSuchElementException If there is no BlobStore config identified by the provided name.
+     * @throws IllegalArgumentException If this configuration can't, for some reason, remove the blob store identified
+     * by the supplied name (for example, the configuration is read-only).
+     */
+    void removeBlobStore(String name) throws NoSuchElementException, IllegalArgumentException;
+
+    /**
+     * Modifies a {@link BlobStoreInfo} in this configuration.
+     * @param info The {@link BlobStoreInfo} that should be used to update an existing BlobStoreInfo with the same
+     * name/id.
+     * @throws NoSuchElementException If there is no BlobStore config identified by the provided name.
+     * @throws IllegalArgumentException If this configuration can't, for some reason, modify the blob store identified
+     * by the supplied name (for example, the configuration is read-only).
+     */
+    void modifyBlobStore(BlobStoreInfo info) throws NoSuchElementException, IllegalArgumentException;
+
+    /**
+     * Retrieve a set of {@link BlobStoreInfo} names in this configuration.
+     * @return A Set of names that can be used to identify each of the {@link BlobStoreInfo}s currently configured.
+     */
+    Set<String> getBlobStoreNames();
+
+    /**
+     * Retrieves the number of {@link BlobStoreInfo}s in this configuration.
+     * @return The number of {@link BlobStoreInfo}s currently configured.
+     */
+    int getBlobStoreCount();
+
+    /**
+     * Retrieves a {@link BlobStoreInfo} from this configuration.
+     * @param name The name that identifies the desired {@link BlobStoreInfo}.
+     * @return An Optional wrapping the desired {@link BlobStoreInfo}, or Empty if it does not exist.
+     */
+    Optional<BlobStoreInfo> getBlobStore(String name);
+
+    /**
+     * Indicates if this configuration can persist the supplied {@link BlobStoreInfo}.
+     * @param info a {@link BlobStoreInfo} to be persisted.
+     * @return True if the supplied {@link BlobStoreInfo} can be persisted, false otherwise. Implementations might
+     * return false if, for example, this configuration is read-only.
+     */
+    boolean canSave(BlobStoreInfo info);
+
+    /**
+     * Replaces a {@link BlobStoreInfo}'s name with a new name.
+     * @param oldName The name that identifies the {@link BlobStoreInfo} to be renamed.
+     * @param newName The name that should replace the current name of the {@link BlobStoreInfo} identified by oldName.
+     * @throws NoSuchElementException If there is no BlobStore config identified by the provided oldName.
+     * @throws IllegalArgumentException If this configuration can't, for some reason, rename the blob store identified
+     * by the supplied name (for example, newName is null or invalid).
+     */
+    void renameBlobStore(String oldName, String newName) throws NoSuchElementException, IllegalArgumentException;
```

Comment:
Looks good. Looking at TileLayerConfiguration, a `containsBlobStore` method _might_ be valuable, but I'm not sure. `getBlobStore` is probably sufficient.

Reply:
@tbarsballe I've added 1`containsBlobStore` to be consistent.

Reply:
added